skbio/stats/ordination/_principal_coordinate_analysis.py

```diff
@@ -6,33 +6,29 @@
 # The full license is in the file COPYING.txt, distributed with this software.
 # ----------------------------------------------------------------------------
 
-from warnings import warn
-
-import pandas as pd
 import numpy as np
+import pandas as pd
+from numpy import dot, hstack
+from numpy.linalg import qr, svd
+from numpy.random import standard_normal
 from scipy.linalg import eigh
+from warnings import warn
 
 from skbio.stats.distance import DistanceMatrix
 from skbio.util._decorator import experimental
 from ._ordination_results import OrdinationResults
-from ._utils import e_matrix, f_matrix
-
-# - In cogent, after computing eigenvalues/vectors, the imaginary part
-#   is dropped, if any. We know for a fact that the eigenvalues are
-#   real, so that's not necessary, but eigenvectors can in principle
-#   be complex (see for example
-#   http://math.stackexchange.com/a/47807/109129 for details) and in
-#   that case dropping the imaginary part means they'd no longer be
-#   so, so I'm not doing that.
+from ._utils import center_distance_matrix_optimized
 
 
 @experimental(as_of="0.4.0")
-def pcoa(distance_matrix):
+def pcoa(distance_matrix, method="eigh", to_dimension=None,
```

Comment:
@gregcaporaso @ebolyen, I'd like to draw your attention to this PR as it expands the API for `pcoa`. 

The motivation for the expansion is to support a heuristic for computing eigenvectors, and builds off an MDS approximation benchmark underway [here](https://github.com/biocore/mds-approximations/). The reason for this work is both performance and necessity: PCoA using `eigh` cannot handle Qiita-sized analyses. We don't know the exact number of samples where it breaks down (as in, produces `nan`, but I'd guess its in the 60-90k sample range. On 100k samples though, the FSVD approach takes only about 2 hours CPU time which is quite impressive.